dhctl/pkg/operations/bootstrap/cluster-bootstrapper.go

```diff
@@ -0,0 +1,799 @@
+	"github.com/deckhouse/deckhouse/dhctl/pkg/system/ssh"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/template"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/terminal"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/terraform"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/util/tomb"
+	"github.com/google/uuid"
+)
+
+const (
+	banner = "" +
+		`========================================================================================
+ _____             _     _                                ______                _ _____
+(____ \           | |   | |                              / _____)              | (_____)
+ _   \ \ ____ ____| |  _| | _   ___  _   _  ___  ____   | /      ____ ____   _ | |  _
+| |   | / _  ) ___) | / ) || \ / _ \| | | |/___)/ _  )  | |     / _  |  _ \ / || | | |
+| |__/ ( (/ ( (___| |< (| | | | |_| | |_| |___ ( (/ /   | \____( ( | | | | ( (_| |_| |_
+|_____/ \____)____)_| \_)_| |_|\___/ \____(___/ \____)   \______)_||_|_| |_|\____(_____)
+========================================================================================`
+
+	bootstrapAbortInvalidCacheMessage = `Create cache %s:
+	Error: %v
+	Probably that Kubernetes cluster was successfully bootstrapped.
+	Use "dhctl destroy" command to delete the cluster.
+`
+	bootstrapPhaseBaseInfraNonCloudMessage = `It is impossible to create base-infrastructure for non-cloud Kubernetes cluster.
+You have to create it manually.
+`
+	bootstrapAbortCheckMessage = `You will be asked for approval multiple times.
+If you are confident in your actions, you can use the flag "--yes-i-am-sane-and-i-understand-what-i-am-doing" to skip approvals.
+`
+	cacheMessage = `Create cache %s:
+	Error: %v
+
+	Probably that Kubernetes cluster was successfully bootstrapped.
+	If you want to continue, please delete the cache folder manually.
+`
+)
+
+// TODO(remove-global-app): Support all needed parameters in Params, remove usage of app.*
+type Params struct {
+	// TODO(dhctl-for-commander): add initial state and on-phase callback param
+
+	ConfigPath              string
+	ResourcesPath           string
+	ResourcesTimeout        string
+	DeckhouseTimeout        time.Duration
+	SSHUser                 string
+	SSHBastionUser          string
+	SSHAgentPrivateKeys     []string
+	PostBootstrapScriptPath string
+	UseTfCache              *bool
+	AutoApprove             *bool
+
+	*client.KubernetesInitParams
+}
+
+type ClusterBootstrapper struct {
+	*Params
+
+	reinitializeSSHPrivateKeys bool
+}
+
+func NewClusterBootstrapper(params *Params) *ClusterBootstrapper {
+	return &ClusterBootstrapper{
+		Params: params,
+	}
+}
+
+// TODO(remove-global-app): Eliminate usage of app.* global variables,
+// TODO(remove-global-app):  use explicitly passed params everywhere instead,
+// TODO(remove-global-app):  applyParams will not be needed anymore then.
+//
+// applyParams overrides app.* options that are explicitly passed using Params struct
+func (b *ClusterBootstrapper) applyParams() error {
+	if b.ConfigPath != "" {
```

Comment:
I hope at the first stage each dhctl launch will be in a separate instance? Otherwise the magic won't work.

I see a comment that we will then get rid of the use of app .* global variables in the code and this is a good way.